### ISSUE TITLE
cannot import name mapmri 

### ISSUE THREAD

**Issue by soichih** [brain-life/app-dipy-MAPMRI#4]
I tried to run this app and ran into following issue.

```
Traceback (most recent call last):
  File "/N/dc2/scratch/brlife/karst-workflows/5a4e98fd3c6325002f5430af/5a4e98fd3c6325002f5430b0/main.py", line 1, in <module>
    from dipy.reconst import mapmri
ImportError: cannot import name mapmri

real	0m5.700s
user	0m0.075s
sys	0m0.303s
```

I recommend making following changes.

1) Remove start/stop/status hook and use the new `main` based execution model. It's documented here > http://www.brainlife.io/warehouse/helloworld
2) The error message seems to be caused by incompatible dipy installed on the resource where this application is running. The best way to overcome dependency issue is to use Docker container. We recommend using this container that's already built (https://hub.docker.com/r/brainlife/dipy) or use it as a base image to install other dependencies needed. Then, `main` can run the main python script with appropriate configuration parameters similar to https://github.com/brain-life/app-dipy/blob/master/main

**Comment by arokem** [comment]
You might want to use a pip install to get newer versions of dipy (rather
than apt-get).

Also, for your docker images, it would be a good idea to create tags. For
example, when a new version of dipy comes out, and you happen to run an
automated build here, the version of dipy will change, but because you only
use the `latest` tag, it will be overwritten and the behavior of your app
might change in inexplicable ways.

**Comment by francopestilli** [comment]
@arokem do you mean tag the repository on github for every docker build or tag on dockerhub? Tag on github seems like a good routine thing to do for each build.

**Comment by arokem** [comment]
Dockerhub

**Comment by soichih** [comment]
@arokem Thanks for the suggestion. I've updated the Dockerfile to install specific version of dipy (via pip). I've also updated the Dockerhub's autobuild config to build specific tags (0.13.0 for now). I am hoping, though, eventually I can use the dipy team's official docker container - at least as a base image.

**Comment by arokem** [comment]
:+1:

That all sounds great. I think that long run we'll want containers that do
particular tasks (workflows) as well.

**Comment by aarya22** [comment]
@soichih Note, the app currently on brain-life is not the correct application. This is the old application that does not take advantage of the Mapmri workflow on dipy. 

Once I locally test the version of the application that I am developing (aarya22/app-dipy-mapmri) I'll pull request it here.

**Comment by soichih** [comment]
@aarya22 Great! Will it be part of 0.14 release maybe?

If there will be a dipy workflow for mapmri, I'd like to suggest it to be part of https://github.com/brain-life/app-dipy-workflows This is a wrapper app that should be able to run all dipy workflow by registering different brainlife app that uses this same app underneath it. We just have to add the new command line inside `main`.

**Comment by arokem** [comment]
We'll try to get the MAPMRI workflow into 0.14. It's pretty close to ready, I think! 

I see that you updated your docker image building. Did that work for you, or are you still getting the import error you mentioned above?

**Comment by aarya22** [comment]
@soichih Not sure exactly how to incorporate this into app-dipy-workflows. We can work on that later? 

Im also not sure we can use this current docker build for the mapmri application. The mapmri pull request should be added to dipy soon and should be in the 0.14 release. It might only necessarily work on that new release.

**Comment by soichih** [comment]
@arokem Yes, the autobuild on dockerhub works great - although it's for brain-life/app-dipy-workflows, not this repo. I believe the same issue still exists with this repo, however, as it uses the dipy installed on the environment. 

@aarya22 brain-life/app-dipy-workflow is tagged with 0.13.0 so we can go ahead with updating the script /  dockerfile to use 0.14 feature - although we can't actually build the container until it's released.

**Comment by aarya22** [comment]
@soichih Guess then we'll have to wait a little bit for the testing of the application. 

Or we could build from dipy directly from GitHub. But that might make things messy as it did before.
